### PR TITLE
Fix import path for bitcask

### PR DESCRIPTION
**Pull request by prologic**
**Description**

Project has moved to [git.mills.io/prologic/bitcask](https://git.mills.io/prologic/bitcask).

For details on why see: https://github.com/prologic

Thank you! 🙇

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by prologic** [comment]
I don't think the CI is failing due to anything I did :/

**Review by mgfeller — COMMENTED** [review]
Hmm...

```
meshery on  migrate_bitcask [?] via 🐹 v1.15.5 took 2s 
❯ make golangci-run   
go run github.com/layer5io/meshkit/cmd/errorutil -d . analyze -i ./helpers -o ./helpers --skip-dirs mesheryctl
go: git.mills.io/prologic/bitcask@v0.3.9: parsing go.mod:
        module declares its path as: github.com/prologic/bitcask
                but was required as: git.mills.io/prologic/bitcask
Makefile:176: recipe for target 'error' failed
make: *** [error] Error 1
```

as well as 
```
meshery on  migrate_bitcask [?] via 🐹 v1.15.5 
❯ go mod tidy
go: git.mills.io/prologic/bitcask@v0.3.9: parsing go.mod:
        module declares its path as: github.com/prologic/bitcask
                but was required as: git.mills.io/prologic/bitcask
```

File: go.mod [hunk 1]
```diff
@@ -47,7 +47,7 @@ require (
 	github.com/nsf/termbox-go v1.1.1
 	github.com/olekukonko/tablewriter v0.0.5
 	github.com/pkg/errors v0.9.1
-	github.com/prologic/bitcask v0.3.9
+	git.mills.io/prologic/bitcask v0.3.9
```
Comment (mgfeller) [inline go.mod:50]: v0.3.14?

File: go.mod [hunk 2]
```diff
@@ -6,7 +6,7 @@ replace (
 	//golang.org/x/sys => golang.org/x/sys v0.0.0-20200826173525-f9321e4c35a6
 	github.com/docker/docker => github.com/moby/moby v17.12.0-ce-rc1.0.20200618181300-9dc6525e6118+incompatible
 	github.com/kudobuilder/kuttl => github.com/layer5io/kuttl v0.4.1-0.20200806180306-b7e46afd657f
-	github.com/prologic/bitcask => git.mills.io/prologic/bitcask v0.3.9
+	git.mills.io/prologic/bitcask => git.mills.io/prologic/bitcask v0.3.9
```
Comment (mgfeller) [inline go.mod:9]: this line can be removed

**Review by mgfeller — CHANGES_REQUESTED** [review]
Thanks! When bumping to v0.3.14, `go mod tidy` was successful (tested locally only).

**Comment by leecalcote** [comment]
Thank you, @prologic!

File: docs/pages/concepts/architecture/database.md [hunk 3]
```diff
@@ -24,5 +24,5 @@ Meshery Database has several kinds of database implementations to support variou
 
 | Component      | Library                               |
 | :------------- | :------------------------------------ |
-| Bitcask        | github.com/prologic/bitcask           |
-| SQLite         | gorm.io/gorm, gorm.io/driver/sqlite   |
\ No newline at end of file
+| Bitcask        | git.mills.io/prologic/bitcask           |
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/database.md:27]:
```suggestion
| Bitcask        | git.mills.io/prologic/bitcask         |
```

Comment (leecalcote) [reply docs/pages/concepts/architecture/database.md:27]: Removing whitespace for purposes of kicking off a new build with refreshed workflow configuration (should fix some of the failed build checks).

**Comment by l5io** [comment]
🚀 Preview for commit b9d68fc32df8236a71cc3cc8e0440b5987112d91 at: https://60f9d01a2051704b7a16620f--meshery-docs-preview.netlify.app

File: go.mod (hunk 1 above)
Comment (prologic) [reply go.mod:50]: Yes! Please. Sorry I was being lazy and did these PR(s) with a shell script en-mass. Please upgrade to 0.3.14 -- We are releasing v1 this weekend 🎉

File: go.mod (hunk 2 above)
Comment (prologic) [reply go.mod:9]: Indeed

**Comment by prologic** [comment]
> Thank you, @prologic!

Quite welcome! Sorry for the breakage there too, I kind of migrated off Github a bit "ass" backwards 🤣

File: go.mod (hunk 1 above)
Comment (mgfeller) [reply go.mod:50]: > Yes! Please. Sorry I was being lazy and did these PR(s) with a shell script en-mass. Please upgrade to 0.3.14 -- We are releasing v1 this weekend 🎉

@prologic ok, cool, thanks for the feedback! Will do. And congrats to v1!

**Comment by l5io** [comment]
🚀 Preview for commit 9edad0b6ba94d267f330386c37864d7c5f2f7ae7 at: https://60fa7b27d5fb278a9cdb1353--meshery-docs-preview.netlify.app

**Comment by mgfeller** [comment]
I don't understand why this is failing - and how to fix it.
The "missing" functions are from `github.com/gofrs/flock v0.8.0`. This module is in `go.sum`. 
I experienced the same problem locally, then added `github.com/gofrs/flock v0.8.0 // indirect` to `go.mod`, then it worked.
Running `go mod tidy` removed this line from `go.mod`, and it still worked...

So, if  `github.com/gofrs/flock v0.8.0 // indirect` is added to  `go.mod`, committed and pushed it might work (haven't tried).
Then, next one to update `go.mod` and doing a `go mod tidy` will cause it to fail again?

Puzzled...

// @kumarabd @Aisuko

**Comment by prologic** [comment]
Is this a problem with our dependency on flock?

**Comment by mgfeller** [comment]
Hmm, not sure my previous comment is correct, I can't reproduce it.

I'm just familiarizing myself with the code and Bitcask. 

Maybe the problem is that we handled locks explicitly, is that necessary? Tried to find some doc about this, but was unsuccessful.

So, we're using e.g.  `locked, err := s.db.TryRLock()` where `db` is `*bitcask.Bitcask`

In v0.3.9, `*flock.Flock` is embedded.
```
type Bitcask struct {
	mu sync.RWMutex

	*flock.Flock
```

No anymore in v0.3.14

```
type Bitcask struct {
	mu         sync.RWMutex
	flock      *flock.Flock
	config     *config.Config
```

**Comment by l5io** [comment]
🚀 Preview for commit f602405fe5fae5dce1b4b52b811907f41d3c7fe6 at: https://60faac161e32220091b5e965--meshery-docs-preview.netlify.app

**Review by mgfeller — APPROVED** [review]

**Comment by kumarabd** [comment]
LGTM

**Comment by prologic** [comment]
> Thanks for the feedback @prologic !
> 
> 
> 
> Hmm, not sure my previous comment is correct, I can't reproduce it.
> 
> 
> 
> I'm just familiarizing myself with the code and Bitcask. 
> 
> 
> 
> Maybe the problem is that we handled locks explicitly, is that necessary? Tried to find some doc about this, but was unsuccessful.
> 
> 
> 
> So, we're using e.g.  `locked, err := s.db.TryRLock()` where `db` is `*bitcask.Bitcask`
> 
> 
> 
> In v0.3.9, `*flock.Flock` is embedded.
> 
> ```
> 
> type Bitcask struct {
> 
> 	mu sync.RWMutex
> 
> 
> 
> 	*flock.Flock
> 
> ```
> 
> 
> 
> Not anymore in v0.3.14
> 
> 
> 
> ```
> 
> type Bitcask struct {
> 
> 	mu         sync.RWMutex
> 
> 	flock      *flock.Flock
> 
> 	config     *config.Config
> 
> ```
> 
> 
> 
> 

Oh! 🤦‍♂️ I don't think that was ever suppose to be part of the public API!

Good thing we hadn't released that as v1 😳

**Comment by prologic** [comment]
Changes LGTM too 👌

**Comment by leecalcote** [comment]
Whoo-hoo!

**Pull request merged**